autometa/common/kmers.py

```diff
@@ -465,7 +465,34 @@ def do_UMAP(n_neighbors=15, n_components=n_components, metric='euclidean'):
         logger.debug(f'embedded.shape {embedded_df.shape} : Written {embedded}')
     return embedded_df
 
-def main(args):
+def main():
+    import argparse
+    import logging as logger
+    logger.basicConfig(
+        format='[%(asctime)s %(levelname)s] %(name)s: %(message)s',
+        datefmt='%m/%d/%Y %I:%M:%S %p',
+        level=logger.DEBUG)
+    skip_desc = '(will skip if file exists)'
+    cpus = mp.cpu_count()
+    parser = argparse.ArgumentParser(description='Count k-mer frequencies of given `fasta`')
+    parser.add_argument('--fasta', help='</path/to/sequences.fna>', required=True)
+    parser.add_argument('--kmers', help=f'</path/to/output/kmers.tsv> {skip_desc}', required=True)
+    parser.add_argument('--size', help='k-mer size', default=5, type=int)
+    parser.add_argument('--normalized', help=f'</path/to/output/kmers.normalized.tsv> {skip_desc}')
+    parser.add_argument('--embedded', help='</path/to/kmers.embedded.tsv>')
+    parser.add_argument('--method', help='embedding method', choices=['TSNE','UMAP'], default='UMAP')
+    parser.add_argument('--n-components', help='<num components of lower dimension manifold>',
+        type=int, default=2)
```

Comment:
This is a bit too jargonny for users. Suggest "Number of dimensions to reduce k-mer frequencies to"